client/state/site-settings/exporter/actions.js

```diff
@@ -83,28 +96,32 @@ export function advancedSettingsFail( siteId, error ) {
  * @param  {Number}   siteId  The ID of the site to export
  * @return {Function}         Action thunk
  */
-export function startExport( siteId ) {
-	return ( dispatch ) => {
+export function startExport( siteId, exportAll = true ) {
+	return ( dispatch, getState ) => {
 		if ( ! siteId ) {
 			return;
 		}
 
 		dispatch( {
 			type: EXPORT_START_REQUEST,
-			siteId: siteId
+			siteId,
+			exportAll,
 		} );
 
+		const state = getState();
+		const advancedSettings = exportAll ? null : prepareExportRequest( state, siteId );
```

Comment:
If prepareExportRequest is called in more than one place, maybe it would make more sense to abstract the exportAll logic in prepareExportRequest.
```
const advancedSettings = prepareExportRequest( state, options );
```